examples/games/flappy_bevy.rs

```diff
@@ -0,0 +1,278 @@
+//! A simplified Flappy Bird but with many birds
+
+use bevy::prelude::*;
+
+use bevy::sprite::collide_aabb::collide;
+use bevy::window::PresentMode;
+
+use rand::Rng;
+
+const CHUNK_SIZE: f32 = 300.0;
+const CAMERA_SPEED: f32 = 120.0;
+const GAME_HEIGHT: f32 = 500.0;
+const SCREEN_HEIGHT: f32 = 1500.0;
+const CLEANUP_X_DIST: f32 = 1500.0;
+const CHAOS_AMOUNT_X: f32 = 250.0;
+const CHAOS_AMOUNT_Y: f32 = 600.0;
+const DRIFT_TO_CENTER_AMOUNT: f32 = 0.01;
+const FLAP_STRENGTH: f32 = 250.0;
+const BIRD_SIZE: f32 = 24.0;
+const BIRD_REPRODUCTION_CHANCE: f32 = 1.0;
+const MAX_BIRDS: usize = 500;
+const GRAVITY: f32 = 400.0;
+
+fn randf() -> f32 {
```

Comment:
I would remove this function; I don't think it makes anything clearer.

Reply:
randf() is used 10 times, so it would result in 10 instances of rand::thread_rng().gen::<f32>() in the example, which seems ugly to me.

Reply:
You should store the `rand::thread_rng()` object in a resource. That design is generally lower overhead, and makes eventually extracting the RNG to control seeds much easier.

Reply:
I don't think that's possible because ThreadRng is not Send

Reply:
Ah, you can use a `NonSend` resource for that then. Although perhaps there's a more sensible RNG initialization strategy to follow.

Reply:
`rand::thread_rng()` should be cheap to call. All it does is take a reference to a thread local storage slot.

Reply:
The larger problem here IMO is introducing uncontrollable gameplay nondeterminism. For a flappy bird game that's fine, but it's not exactly the pattern I want to be teaching new users.

Reply:
Maybe store a `SmallRng` then? I believe that one is Send+Sync.

Reply:
for now I have used `random::<f32>()`